### PR TITLE
[gl] Generate GLSL with SPIRV-Cross

### PR DESCRIPTION
**Pull request by grovesNL**
Adds SPIRV-Cross GLSL compilation and fixed some trivial warnings. Most of the implementation is taken from DX12 backend for now.

### CODE REVIEW

**Comment by grovesNL** [comment]
Thanks, I’ll squash soon.

We weren’t building SPIRV-Cross on Linux before. We need to add gcc or clang to build it.

File: src/backend/gl/src/device.rs [hunk 1]
```diff
@@ -241,19 +293,31 @@ impl d::Device<B> for Device {
...
+                    let attach_shader = |point_maybe: Option<&pso::EntryPoint<B>>, stage: pso::Stage| {
                         if let Some(point) = point_maybe {
                             assert_eq!(point.entry, "main");
-                            unsafe { gl.AttachShader(name, point.module.raw); }
+                            let raw = match *point.module {
+                                n::ShaderModule::Raw(raw) => raw,
+                                n::ShaderModule::Spirv(ref spirv) => {
+                                    let mut ast = self.parse_spirv(spirv).unwrap();
+                                    let spirv = self.translate_spirv(&mut ast, glsl::Version::V4_10).unwrap();
+                                    println!("Generated:\n{:?}", spirv);
```
Comment (msiglreith) [inline src/backend/gl/src/device.rs]: nit: `debug!` or `info!`

**Comment by kvark** [comment]
thanks! bors r+

**Comment by grovesNL** [comment]
Seems like bors didn't see this?

**Pull request merged**